nomos-sdp/src/lib.rs

```diff
@@ -39,6 +41,18 @@ pub enum ServiceType {
     GenericRestaking,
 }
 
+impl fmt::Display for ServiceType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let service_str = match self {
+            Self::BlendNetwork => "BlendNetwork",
+            Self::DataAvailability => "DataAvailability",
+            Self::ExecutorNetwork => "ExecutorNetwork",
+            Self::GenericRestaking => "GenericRestaking",
+        };
+        write!(f, "{service_str}")
+    }
+}
+
```

Comment:
Can't this be derived? 🤔

Reply:
Hmm, maybe I am missing something. How do I derive the Display, using `strum` or some lib? Sorry if it's a noob question

Reply:
For example:
```
if let Ok(providers) = providers {
                                if tx.send(providers).is_err() {
                                    tracing::error!("Error sending initial membership snapshot for service type: {}", service_type);
                                } else if result_sender.send(stream).is_err() {
                                    tracing::error!("Error sending finalized updates receiver for service type: {}", service_type);
                                }
                            } else {
                                tracing::error!("Failed to get latest providers for service type: {}", service_type);
                            }

 ```

Reply:
In that case you can use the `Debug` version of it directly:
```
tracing::error!("Error sending initial membership snapshot for service type: {:?}", service_type);
```
Which is basically what you implemented anyway iirc.

Reply:
Makes sense, fixed.